app/lib/actions/katello/organization/create.rb

```diff
@@ -3,30 +3,33 @@ module Katello
     module Organization
       class Create < Actions::EntryAction
         def plan(organization)
-          organization.setup_label_from_name
-          organization.create_library
-          organization.create_anonymous_provider unless organization.anonymous_provider
-          organization.create_redhat_provider unless organization.redhat_provider
-          cp_create = nil
+          return if organization.created_in_katello
 
-          organization.save!
+          creator = ::Katello::OrganizationCreator.new(organization)
+          creator.seed!
 
           sequence do
-            cp_create = plan_action(Candlepin::Owner::Create,
-                                    label:  organization.label,
-                                    name: organization.name)
-            plan_action(Katello::Environment::LibraryCreate, organization.library)
-          end
-          if cp_create
-            action_subject organization, label: cp_create.output[:response][:key]
-          else
-            action_subject organization
+            unless organization.candlepin_owner_exists?
+              plan_action(Candlepin::Owner::Create,
+                label: organization.label,
+                name: organization.name
+              )
+            end
+
+            if creator.library_view.cp_environment_id(creator.library_environment).nil?
+              plan_action(Katello::ContentView::AddToEnvironment, creator.library_cvv, creator.library_environment)
+            end
           end
+
           plan_self(:organization_id => organization.id)
         end
 
         def run
-          ::Organization.find_by(:id => input[:organization_id]).try(:load_debug_cert)
+          org = ::Organization.find_by(:id => input[:organization_id])
+          # do we need to create the debug cert here?
+          # why was it written with try...
+          org.try(:load_debug_cert)
```

Comment:
I don't know why `.try` was used, but I think at this point `org` should not be undefined and we deserve any NoMethodErrors that causes.

(anyway I'm partial to `&.` anyway)

Reply:
I'm leaning that way too. Perhaps @jlsherrill will take us to school here...